volatility3/framework/plugins/regexscan.py

```diff
@@ -46,38 +46,63 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             ),
         ]
 
-    def _generator(self, regex_pattern):
-        regex_pattern = bytes(regex_pattern, "UTF-8")
-        vollog.debug(f"RegEx Pattern: {regex_pattern}")
-
+    def _generator(self, compiled_pattern, raw_pattern, maxsize):
+        vollog.debug(f"RegEx Pattern: {raw_pattern}")
         layer = self.context.layers[self.config["primary"]]
+
         for offset in layer.scan(
-            context=self.context, scanner=scanners.RegExScanner(regex_pattern)
+            context=self.context, scanner=scanners.RegExScanner(raw_pattern)
         ):
-            result_data = layer.read(offset, self.MAXSIZE_DEFAULT, pad=True)
+            result_data = layer.read(offset, maxsize, pad=True)
 
             # reapply the regex in order to extract just the match
-            regex_result = re.match(regex_pattern, result_data)
+            regex_result = compiled_pattern.search(result_data)
```

Comment:
The need for this is somewhat annoying, but probably can't be improved at this point without breaking the regexscanner...